webapp/channels/src/utils/url.tsx

```diff
@@ -168,6 +167,24 @@ export function validateChannelUrl(url: string, intl?: IntlShape): Array<React.R
     return errors;
 }
 
+// Returns true when the URL should not trigger any external requests.
+// Currently returns true only for permalinks
+export function doesNotTriggerExternalRequests(url: string, siteURL?: string): boolean {
+    if (siteURL && siteURL !== '') {
+        let standardSiteURL = siteURL;
+        if (standardSiteURL[standardSiteURL.length - 1] === '/') {
+            standardSiteURL = standardSiteURL.substring(0, standardSiteURL.length - 1);
+        }
+        if (!url.startsWith(standardSiteURL)) {
+            return false;
+        }
+        const afterSiteURL = url.substring(standardSiteURL.length);
+        const regexp = new RegExp('^/[0-9a-z_-]{1,64}/pl/[0-9a-z_-]{26}$');
```

Comment:
0/5 this could be a regex literal now instead of calling the constructor

Reply:
Or we can move this as a constant at global level, and that way build it just once.